app/app_main.c

```diff
@@ -3645,23 +3670,60 @@ static ACVP_RESULT app_kas_ecc_handler(ACVP_TEST_CASE *test_case)
         printf("Peerkey failed\n");
         return ACVP_CRYPTO_MODULE_FAIL;
     }
-    if (!EC_KEY_generate_key(ec)) {
-        EC_POINT_free(peerkey);
-        EC_GROUP_free(group);
-        BN_free(cx);
-        BN_free(cy);
-        printf("EC_KEY_generate_key failed\n");
-        return ACVP_CRYPTO_MODULE_FAIL;
+    if (tc->test_type == ACVP_KAS_ECC_TT_VAL) {
+        if (!BN_hex2bn(&ix, (char *)tc->pix)) {
+            EC_GROUP_free(group);
+            BN_free(cx);
+            BN_free(cy);
```

Comment:
Down to preference, but goto cleanup and then the memory can be freed there. Saves redundant lines for each for each of these error statements. Up to you :)

Reply:
I'll commit that change with the second half of the TOTP fix that I missed.
You'll have to re-approve I think.